packages/cli/src/commands/integration-resource/create-threshold.ts

```diff
@@ -0,0 +1,366 @@
+  prepaymentInfo: InstallationBalancesAndThresholds;
+  minimum: number;
+  spend: number;
+  limit: number;
+  skipConfirmWithYes: boolean;
+}) {
+  const existingThreshold = props.prepaymentInfo.thresholds.find(
+    threshold => threshold.resourceId === props.resource.externalResourceId
+  );
+  const existingBalance = props.prepaymentInfo.balances.find(
+    balance => balance.resourceId === props.resource.externalResourceId
+  );
+
+  return handleUpdateThreshold({
+    client: props.client,
+    resource: props.resource,
+    minimum: props.minimum,
+    spend: props.spend,
+    limit: props.limit,
+    existingThreshold,
+    existingBalance,
+    skipConfirmWithYes: props.skipConfirmWithYes,
+    isInstallationLevel: false,
+  });
+}
+
+async function updateThresholdForInstallation(props: {
+  client: Client;
+  resource: Resource;
+  prepaymentInfo: InstallationBalancesAndThresholds;
+  minimum: number;
+  spend: number;
+  limit: number;
+  skipConfirmWithYes: boolean;
+}) {
+  const existingThreshold = props.prepaymentInfo.thresholds.find(
+    threshold => threshold.resourceId === undefined
+  );
+  const existingBalance = props.prepaymentInfo.balances.find(
+    balance => balance.resourceId === undefined
+  );
+
+  return handleUpdateThreshold({
+    client: props.client,
+    resource: props.resource,
+    minimum: props.minimum,
+    spend: props.spend,
+    limit: props.limit,
+    existingThreshold,
+    existingBalance,
+    skipConfirmWithYes: props.skipConfirmWithYes,
+    isInstallationLevel: true,
+  });
+}
+
+async function handleUpdateThreshold(props: {
+  client: Client;
+  resource: Resource;
+  minimum: number;
+  spend: number;
+  limit: number;
+  existingThreshold?: PrepaymentCreditThreshold;
+  existingBalance?: CreditWithAmount;
+  skipConfirmWithYes: boolean;
+  isInstallationLevel: boolean;
+}) {
+  if (props.resource.billingPlan?.type !== 'prepayment') {
+    output.log(
+      `The resource ${chalk.bold(props.resource.name)} is not a prepayment-based resource.`
+    );
+    return 0;
+  }
+
+  if (!props.resource.product?.integrationConfigurationId) {
+    output.log(
+      `The resource ${chalk.bold(props.resource.name)} does not have an integration configuration.`
+    );
+    return 0;
+  }
+
+  const entityTextReference = props.isInstallationLevel
+    ? `installation ${chalk.bold(props.resource.product?.name)}`
+    : `resource ${chalk.bold(props.resource.name)}`;
+  if (props.existingThreshold) {
+    const shouldOverwriteThreshold =
+      props.skipConfirmWithYes ||
+      (await props.client.input.confirm(
+        `The ${entityTextReference} already has a threshold. (minimum: $${props.existingThreshold.minimumAmountInCents / 100}, spend: $${props.existingThreshold.purchaseAmountInCents / 100}, limit: ${props.existingThreshold.maximumAmountPerPeriodInCents ? `$${props.existingThreshold.maximumAmountPerPeriodInCents / 100}` : 'none set'}). Do you want to overwrite it?`,
+        true
+      ));
```

Comment:
> [!NOTE]
> Split to make this easier to read (having the promise resolution directly in a ternary is what is bugging me here I think)

Reply:
For simplicity I'm using this to only call the confirm function if they haven't skipped confirmation, but resolving to the same constant for logic simplicity.